src/patternfly/components/List/list.scss

```diff
@@ -22,6 +37,23 @@
     list-style: var(--pf-c-list--ul--ListStyle);
   }
 
+  .pf-c-list__item-main {
+    display: flex;
+    align-items: center;
+  }
+
+  .pf-c-list__item-icon {
+    display: flex;
+    margin-right: var(--pf-c-list__item-icon--MarginRight);
```

Comment:
Icon width can vary slightly, a more bulletproof implementation would be to use `grid` and `column-gap` instead of `margin-right`